{{cookiecutter.project_slug}}/clients/react/src/pages/SignUp.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { useState, useContext } from "react"
+import {
+  Heading,
+  Input,
+  Text,
+  FormControl,
+  FormErrorMessage,
+  Box,
+  HStack,
+  Link,
+} from "@chakra-ui/react"
+import { useMutation } from "@apollo/client"
+import { CREATE_USER, LOG_IN } from "../utils/mutations"
+import { useForm } from "react-hook-form"
+import { useNavigate } from "react-router-dom"
+import { AuthContext } from "../utils/auth"
+
+interface FormValues {
+  email: string
+  password: string
+  firstName: string
+  lastName: string
+}
+
+export default function SignUp() {
+  const [confirmPassword, setConfirmPassword] = useState("")
+  const [passwordMatch, setPasswordMatch] = useState(true)
+  const [error, setError] = useState("")
+  const { updateToken } = useContext(AuthContext)
+
+  let navigate = useNavigate()
+
+  const { register, handleSubmit } = useForm<FormValues>()
+
+  const [logIn] = useMutation(LOG_IN, {
+    onCompleted: (data: any) => {
+      localStorage.setItem("auth-token", data.tokenAuth.token)
+      updateToken(data.tokenAuth.token)
+      navigate("/home")
+    },
+    onError: (error: any) => {},
```

Comment:
@jdephil can we display this error in the UI also?

Reply:
@jdephil Hmm good question. There shouldn't be an issue between signing up and logging in here, but if there were, I think it's best practice to display the error even if we redirect the user to log in